### ISSUE TITLE
"Last zone sync stage validation failed" warning isn't cleared after successful sync

### ISSUE THREAD

**Issue by PinkFrog** [Icinga/icinga2#7642]
## Describe the bug

After deleting a zone I had some issues with a zone sync and I got the warning ```Last zone sync stage validation failed``` on primary **and** secondary master. After solving the problems with the config, the warning disappeared on the secondary master **but not** at the primary master. 

## To Reproduce

Unfortunatley I have no option to provide a live example. 

Version Overview: 
OS: Debian 10
Icinga2 r2.11.2-1
Icinga Web 2 2.7.3
Director 1.7.1

Configuration of the Setup: 
1. HA Master Setup configured like it is described in the documentation https://icinga.com/docs/icinga2/latest/doc/06-distributed-monitoring/#high-availability-master-with-agents 
2. Created a DMZ zone and endpoint in the zones.conf on both masters.
3. Imported the configuration with a director kickstart to the director.  
4. Imported host configurations from a JSON file from our test environment. 
5. Decided to work without a satellite. 
6. Reconfigured all hosts of the satellite zone.
7. Deleted the zone and endpoint from the zones.conf on both masters. 
8. Done again a kickstart. 
9. Having some problems with the configuration -> the warning appeared the first time on both masters. 
10. Solved the problems and the warning disappeared on the secondary master. 
11. Primary master still shows the warning, even after make some other changes in config which could be deployed. 

Yesterday I got the hint to rename the service of the check to trigger a new entry in ido without any effect. Later I dropped the hole ido but this also doesn't took effect, the warning is still there. 

## Expected behavior

I expected to only see this warning while the master zone can't be synchronized. 

## Screenshots

master1: 
![image](https://user-images.githubusercontent.com/9430670/69316165-85294e00-0c38-11ea-8bd7-05b750274d9a.png)

master2: 
![image](https://user-images.githubusercontent.com/9430670/69316303-d46f7e80-0c38-11ea-9b74-a85124b22866.png)

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`):
```icinga2 - The Icinga 2 network monitoring daemon (version: r2.11.2-1)

Copyright (c) 2012-2019 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: Debian GNU/Linux
  Platform version: 10 (buster)
  Kernel: Linux
  Kernel version: 4.19.0-6-amd64
  Architecture: x86_64

Build information:
  Compiler: GNU 8.3.0
  Build host: runner-LTrJQZ9N-project-298-concurrent-0

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var
```

* Operating System and version:
```Debian GNU/Linux 10``` 

* Enabled features (`icinga2 feature list`):
  * master1:
```Disabled features: compatlog debuglog elasticsearch gelf influxdb livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker command graphite ido-mysql mainlog notification
```
  * master2:
```Disabled features: compatlog debuglog elasticsearch gelf influxdb livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker command graphite ido-mysql mainlog notification
```

* Icinga Web 2 version and modules (System - About):
![image](https://user-images.githubusercontent.com/9430670/69316523-4c3da900-0c39-11ea-810e-37df7e217d8d.png)

* Config validation (`icinga2 daemon -C`):
  * master1:
```[2019-11-21 08:31:41 +0100] information/cli: Icinga application loader (version: r2.11.2-1)
[2019-11-21 08:31:41 +0100] information/cli: Loading configuration file(s).
[2019-11-21 08:31:41 +0100] information/ConfigItem: Committing config item(s).
[2019-11-21 08:31:41 +0100] information/ApiListener: My API identity: master1
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 FileLogger.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 4 NotificationCommands.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 NotificationComponent.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 3 HostGroups.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 15 Hosts.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 Comment.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 GraphiteWriter.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 15 Zones.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 14 Endpoints.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 ExternalCommandListener.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 2 ApiUsers.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 310 UserGroups.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 ApiListener.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 1 IdoMysqlConnection.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 244 CheckCommands.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 2 TimePeriods.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 549 Users.
[2019-11-21 08:31:41 +0100] information/ConfigItem: Instantiated 37 Services.
[2019-11-21 08:31:41 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2019-11-21 08:31:41 +0100] information/cli: Finished validating the configuration file(s).
```

  * master2: 
```[2019-11-21 08:32:08 +0100] information/cli: Icinga application loader (version: r2.11.2-1)
[2019-11-21 08:32:08 +0100] information/cli: Loading configuration file(s).
[2019-11-21 08:32:08 +0100] information/ConfigItem: Committing config item(s).
[2019-11-21 08:32:08 +0100] information/ApiListener: My API identity: master2
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 FileLogger.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 4 NotificationCommands.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 NotificationComponent.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 3 HostGroups.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 15 Hosts.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 Comment.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 GraphiteWriter.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 15 Zones.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 14 Endpoints.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 ExternalCommandListener.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 310 UserGroups.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 ApiListener.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 1 IdoMysqlConnection.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 244 CheckCommands.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 2 TimePeriods.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 549 Users.
[2019-11-21 08:32:08 +0100] information/ConfigItem: Instantiated 37 Services.
[2019-11-21 08:32:08 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2019-11-21 08:32:08 +0100] information/cli: Finished validating the configuration file(s).
```

* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.
  * master1: 
```/*
 * Generated by Icinga 2 node setup commands
 * on 2019-10-29 16:51:45 +0100
 */

object Endpoint "master1.domain.com" {
        host = "master1.domain.com" // That's us
}

object Endpoint "master2.domain.com" {
        host = "master2.domain.com" // Actively connect to the secondary master
}

object Zone "master" {
        endpoints = [ "master1.domain.com", "master2.domain.com" ]
}

/*object Endpoint "dmz.domain.com" {
}*/

/*object Zone "dmz.domain.com" {
        endpoints = [ "dmz.domain.com" ]
        parent = "master"
}*/

/* sync global commands */
object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```

  * master2:
```/*
 * Generated by Icinga 2 node setup commands
 * on 2019-10-29 18:14:27 +0100
 */

object Endpoint "master2.domain.com" {
        host = "master2.domain.com"        // That's us
}

object Endpoint "master1.domain.com" {
        host = "master1.domain.com" // Actively connect to the secondary master
}

object Zone "master" {
        endpoints = [ "master1.domain.com", "master2.domain.com" ]
}

/*object Endpoint "dmz.domain.com" {
}*/

/*object Zone "dmz.domain.com" {
        endpoints = [ "dmz.domain.com" ]
        parent = "master"
}*/

/* sync global commands */
object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
## Additional context

Deleting /var/lib/icinga2/api/{packages,zones,zones-stage}/* does not trigger a sync from master2. Seems like there is only a sync from actions from the director. 

Hope that this is enough information, I don't have any experience opening an issue.

**Comment by dnsmichi** [comment]
@htriem please try to reproduce this with your master-master setup, no need for the director here.

**Comment by jeevanm87** [comment]
I have one master and 2 Icinga instances (zones). I have upgraded 1 of the node to v2-11.1 and then stopped sync with master. 
root@zone02 ]# wc -l /var/lib/icinga2/api/zones/satzone02/director/hosts.conf
7696 /var/lib/icinga2/api/zones/satzone02/director/hosts.conf
agent_endpoints.conf
agent_zones.conf
hosts.conf
services.conf

[root@zone01 ]# wc -l /var/lib/icinga2/api/zones/satzone01/director/hosts.conf
17643 /var/lib/icinga2/api/zones/satzone01/director/hosts.conf
 hosts.conf
 host_templates.conf
 service_apply.conf
 services.conf

I need this to fix first instead upgrading all master and nodes to 2-11.1. 
https://icinga.com/docs/icinga2/latest/doc/16-upgrading-icinga-2/#upgrading-to-v211

**Comment by dnsmichi** [comment]
@jeevanm87 This isn't really related to this issue, right?

**Comment by jeevanm87** [comment]
I have the same error: 
Plugin Output
Remote Icinga instance 'zone02.net' is not connected to 'zone01.net'

**Comment by noqqe** [comment]
Having the same issue

```
Icinga 2 has been running for 47 minutes and 6 seconds. Version: r2.11.2-1; Last zone sync stage validation failed at 2020-01-09 11:28:02 +0100
``` 

Not running any replication  setup. Client version  2.11.2-1.buster 500

**Comment by Al2Klimov** [comment]
@dnsmichi That failed state is eventually cleared by `ApiListener#ConfigUpdateHandler()` on config change. That can't happen if config is not accepted – and doesn't happen if **we**, the primary master, sent the `config::Update` *to the other one*, right?

**Comment by PinkFrog** [comment]
Sorry, I wasn‘ at work the last werks, I will test it next week and will make reply.

**Comment by PinkFrog** [comment]
I have took a look on it. My actual problem is, that the problem only occures in our production system so I'm a littlebit worried to test it there. I will need some more time to find a good time to test it.

**Comment by wolf-in-sheeps-clothing** [comment]
Hi,

I set up a test system during the days to test icingadb, v 2.12, web 2.8, etc. and at the same time I also played with the powershell tool.

My test system is an Ubuntu 18.04 (the master) and a Win2012R2 (the agent is running here, installed with the Powershell Kickstart). Later I updated icinga agent also to 2.12.

I also got the error after I checked the icinga check against the agent.

Of course, I fixed the error on the Win Server, so that the agent started without errors (according to the log).

Now my hope was that the check would get OK, but the warning remained.

The solution was to delete the icinga2.state file on the Win Server in C:\ProgramData\icinga2\var\lib\icinga2  (I renamed it first to icinga2.state.bak, later I deletet the .bak file :-) ).

Then run the check again and everything was OK.
So it seems, that a fine zone sync doesn't overwrite the old bad sync in this file.

Maybe that helps for the solution.

**Comment by Wintermute2k6** [comment]
ref/NC/689456

**Comment by lippserd** [comment]
ref/IP/26897

**Comment by N-o-X** [comment]
Closed by #8225
